libs/backend/src/system_call/export_logs_to_usb.ts

```diff
@@ -1,28 +1,114 @@
-import { Result, err, ok } from '@votingworks/basics';
+import { Result, err, ok, throwIllegalValue } from '@votingworks/basics';
 import { UsbDrive } from '@votingworks/usb-drive';
 import * as fs from 'fs/promises';
 import { join } from 'path';
 
-import { LogEventId, Logger } from '@votingworks/logging';
+import {
+  LogEventId,
+  LogExportFormat,
+  Logger,
+  buildCdfLog,
+  filterErrorLogs,
+} from '@votingworks/logging';
+import { createReadStream, createWriteStream } from 'fs';
+import { pipeline } from 'stream/promises';
+import { createGunzip, createGzip } from 'zlib';
+import { dirSync } from 'tmp';
 import { execFile } from '../exec';
 
 const LOG_DIR = '/var/log/votingworks';
+const COMPRESSED_VX_LOGS_NAME_REGEX = 'vx-logs.log-[0-9]*.gz';
 
 /** type of return value from exporting logs */
 export type LogsResultType = Result<
   void,
-  'no-logs-directory' | 'no-usb-drive' | 'copy-failed'
+  | 'no-logs-directory'
+  | 'no-usb-drive'
+  | 'copy-failed'
+  | 'cdf-conversion-failed'
+  | 'error-filtering-failed'
 >;
 
+async function convertLogsToCdf(
+  logDir: string,
+  outputDir: string,
+  logger: Logger,
+  machineId: string,
+  codeVersion: string
+): Promise<void> {
+  const files = await fs.readdir(logDir);
+
+  // Create CDF for vx-logs.log
+  if (files.includes('vx-logs.log')) {
+    await pipeline(
+      createReadStream(join(logDir, 'vx-logs.log'), 'utf8'),
+      (inputStream: AsyncIterable<string>) =>
+        buildCdfLog(logger, inputStream, machineId, codeVersion),
+      createWriteStream(join(outputDir, 'vx-logs.cdf.log'))
+    );
+  }
+
+  // Create CDF for all compressed vx-logs files
+  for (const file of files) {
+    if (file.match(COMPRESSED_VX_LOGS_NAME_REGEX)) {
+      const cdfFileName = file.replace('vx-logs', 'vx-logs.cdf');
+      await pipeline(
+        createReadStream(join(logDir, file)),
+        createGunzip(),
+        (inputStream: AsyncIterable<string>) =>
+          buildCdfLog(logger, inputStream, machineId, 'DEV'),
```

Comment:
is `'DEV'` here meant to be `codeVersion` like above?

Reply:
yes good catch